library_management_cookiecutter/library/views.py

```diff
@@ -0,0 +1,243 @@
+from django.core import serializers
+from django.conf import settings
+from django.db.models import Count
+from django.utils import timezone
+import csv
+from django.core.mail import EmailMessage
+import zipfile
+import os
+import pyminizip
+from io import StringIO
+
+class Home(TemplateView):
+    template_name = "home.html"
+
+
+class Login(LoginView):
+    """login class"""
+
+    template_name = "login.html"
+
+    def post(self, request, *args, **kwargs):
+        username = request.POST.get("username", None)
+        password = request.POST.get("password", None)
+        user = authenticate(username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return JsonResponse({"message": "success"})
+        return JsonResponse({"message": "username and password not match."}, status=400)
+
+
+class Logout(LogoutView):
+    """logout class"""
+
+    pass
+
+
+class CreateUser(CreateView):
+    template_name = "create_user.html"
+    form_class = UserForm
+
+    def post(self, request, *args, **kwargs):
+        """create user post request"""
+        user_form = self.form_class(request.POST or None)
+
+        if user_form.is_valid():
+            user = user_form.save()
+            user.save()
+            user_role = user.role
+            group = Group.objects.get(name=user_role)
+            user.groups.add(group.id)
+            messages.success(self.request, "successfully register.")
+            return JsonResponse({"message": "success"})
+        return JsonResponse({"message": user_form.errors}, status=400)
+
+class AddBooks(LoginRequiredMixin, MyCustomPermissions, CreateView):
+    """add book"""
+
+    login_url = "login"
+    template_name = "add_book.html"
+    form_class = AddBook
+    permission_required = {"GET": ["library.add_book"]}
+
+    def post(self, request, *args, **kwargs):
+        book_form = self.form_class(request.POST, request.FILES)
+        if book_form.is_valid():
+            book = book_form.save(commit=False)
+            book.available_quantity = book_form.cleaned_data.get("quantity")
+            book.save()
+            messages.success(request, "successfully add book.")
+            return JsonResponse({"message": "success"})
+        return JsonResponse({"message": book_form.errors}, status=400)
+
+
+class SuccessMessage(TemplateView):
+    """ success class"""
+
+    template_name = "successpage.html"
+
+
+class BookList(FormView):
+    """ book list show """
+    login_url = "login"
+    model = Book
+    template_name = "book_list.html"
+    form_class = AsignBook
+
+    def get(self, request, *args, **kwargs):
+        if request.META.get("HTTP_X_REQUESTED_WITH") == "XMLHttpRequest":
+            result = dict()
+            data_list = []
+            result["status"] = "success"
+            for books in Book.objects.all():
+                total_books_assign = AssignedBook.objects.filter(book = books,is_deleted = False).count()
```

Comment:
why are you using different models name  in same class name ?.. use different class...